amplify/backend/function/expressServer/src/controllers/auth.js

```diff
@@ -0,0 +1,164 @@
+      return next(new Error('Email address not associated with any account'));
+    }
+
+    //Generate and set password reset token
+    const token = await user.generatePasswordReset();
+
+    user = await user.save();
+
+    let link = host + '/auth/reset/' + user.resetPasswordToken;
+
+    var apiInstance = new SibApiV3Sdk.TransactionalEmailsApi();
+
+    var sendSmtpEmail = new SibApiV3Sdk.SendSmtpEmail(); // SendSmtpEmail | Values to send a transactional email
+
+    const sender = {
+      email: process.env.FROM_EMAIL,
+      name: 'Caolan Power',
+    };
+    const receivers = [
+      {
+        email: user.email,
+      },
+    ];
+    try {
+      let response = await apiInstance.sendTransacEmail({
+        sender,
+        to: receivers,
+        subject: 'Password change request',
+        textContent: `Hi ${user.username} \n 
+        Please click on the following link ${link} to reset your password. \n\n 
+        If you did not request this, please ignore this email and your password will remain unchanged.\n`,
+      });
+      res.status(200).json({
+        message: 'A reset email has been sent to ' + user.email + '.',
+      });
+      return;
+    } catch (error) {
+      console.log(error);
+    }
+  } catch (error) {
+    res.status(500);
+    res.errormessage = error.message;
+    return next(new Error('Server error in password recovery'));
+  }
+};
+
+// @route POST api/auth/reset
+// @desc Reset Password
+// @access Public
+const resetPassword = async (req, res, next) => {
+  let user = await User.findOne({
+    resetPasswordToken: req.params.token,
+    resetPasswordExpires: { $gt: Date.now() },
+  });
+
+  if (!user) {
+    res.status(401);
+    res.errormessage = 'Password reset token is invalid or has expired';
+    return next(new Error('Password reset token invalid or has expired'));
+  }
+
+  //Set the new password
+  user.password = req.body.password;
+  user.resetPasswordToken = undefined;
+  user.resetPasswordExpires = undefined;
+
+  try {
+    user = await user.save();
+  } catch (error) {
+    res.status(500);
+    res.errormessage = err.message;
+    return next(new Error(err.message));
+  }
+
+  var apiInstance = new SibApiV3Sdk.TransactionalEmailsApi();
+
+  var sendSmtpEmail = new SibApiV3Sdk.SendSmtpEmail(); // SendSmtpEmail | Values to send a transactional email
+
+  const sender = {
+    email: process.env.FROM_EMAIL,
+    name: 'Caolan Power',
+  };
+  const receivers = [
+    {
+      email: user.email,
+    },
+  ];
+  try {
+    let response = await apiInstance.sendTransacEmail({
+      sender,
+      to: receivers,
+      subject: 'Your password has been changed',
+      textContent: `Hi ${user.username} \n 
+      This is a confirmation that the password for your account ${user.email} has just been changed.\n`,
+    });
+    res.status(200).json({
+      message: 'A reset email has been sent to ' + user.email + '.',
+    });
+    return;
+  } catch (error) {
+    console.log(error);
```

Comment:
Again think you may need to refactor code a little, if I enter this nested try/catch and there is an error the server will hang as there is no response being returned just a console log. Code gets bit messy with these nested try/catchs. Try to maybe structure entire function if you can. 
```
try{
  <Now in here if there is particular errors use if loop>
   if req.token = "undefined" -> Bad data response 
   ....
   ....
   ....
}catch(error){
  <serverError>
}
```